db/migrations/20170821164207_add_name_to_movies.js

```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+exports.up = function (Knex, Promise) {
+  return Knex.schema.table('movies', (table) => {
+    table.text('name');
+  })
+  .then(() => {
+    return Knex.raw('ALTER TABLE movies ALTER COLUMN title DROP NOT NULL');
+  });
+};
+
+exports.down = function (Knex, Promise) {
+  return Knex.schema.table('movies', (table) => {
+    table.dropColumn('name');
+  })
+  .then(() => {
+    return Knex.raw('ALTER TABLE movies ALTER COLUMN title SET NOT NULL');
```

Comment:
Here and throughout this PR, there are several `ALTER COLUMN` commands. This is fine on small tables, but keep in mind on the larger tables, these can lock the entire table for a long time while it checks to make sure every row has an OK value. 

After I made this mistake, I got to write a little write-up on the matter. It explains how we handle altering table constraints in a way that minimizes the amount of time a table is locked at the minimal level of lock. 

[Notion Doc](https://www.notion.so/lob/Writing-Migrations-15caf9dd14f74c47b498fd7bee8a7b51)